assets/js/fake-data.test.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Tests for faker.js data generation.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import faker from 'faker';
+
+describe( 'Fake data generation', () => {
+	it( 'should be consistent based on the seed data', () => {
+		// This ensures all data created by faker throughout tests is consistent
+		// between test runs.
+		//
+		// This prevents any issues with some test data, when entirely random,
+		// causing issues generating random data when used for things like unique
+		// IDs, React `key` props, etc.
+		//
+		// If this test is failing, you either changed `knownSafeSeed` from `500`
+		// or called `faker.seed()` with a different seed and did not restore the
+		// original seed using `faker.seed( faker.originalSeed )` in the cleanup
+		// for your test.
```

Comment:
@tofumatt is the cleanup even necessary anymore since this is will be set before every test now?

Reply:
Good question! Maybe not, I think this might be more relevant if we did this as a `beforeAll` hook.

Which I suppose is more efficient, but also more error-prone. We should keep this around to prevent people from messing with the seed data (it will make for a much more obvious test failure than other ones), but I'll adjust the comment about cleanup if it isn't necessary.

Reply:
Looks like it doesn't matter:

<img width="844" alt="CleanShot 2022-04-11 at 16 40 31" src="https://user-images.githubusercontent.com/90871/162778453-30936076-01b1-47af-8aee-76efd67e530c.png">

Reply:
I've updated the comment to tweak the wording here, but I think there's value in keeping the test around 🙂